#163345966 Build out GET REST Endpoints for all Offices (CRUD)

amaechi-chuks opened the pull request:
**What does this PR do?**
This PR sets up the GET /api/v1/offices endpoint

**Description of tasks**
- Ensure that the required parameters are provided
- Validate the provided parameters
- Return resource if all provided parameters satisfy the required specifications

**How can this PR be tested**
- Clone this repo, cd into the folder and perform (npm run start:dev)
- Using Postman, set the request method GET
- Specify the endpoint /api/v1/offices
- Click the send button

**Link to relevant story**
https://www.pivotaltracker.com/story/show/163345966

amaechi-chuks committed:
feat[API endpoints #163345966]Set up GET offices endpoint for all
offices
[Finishes|Deliver #163345966]

amaechi-chuks committed:
feature[API endpoint] Set up GET REST API endpoint(CRUD)

yomigeek commented on server/app.js:
```diff
@@ -16,6 +16,10 @@ app.use(bodyParser.urlencoded({ extended: true }));
 const port = process.env.PORT || 6000;
 
+app.get('/', (req, res) => res.status(200).json({
+  message: 'Welcome to Politico',
```
Please check the project document for the correct response message format.

amaechi-chuks replied:
Acknowledged with thanks.
I have implemented the changes based on the correct message format

yomigeek commented on server/controllers/officeControllers.js:
```diff
@@ -4,7 +4,7 @@ import officesDb from '../models/officeModels';
  * Class representing OfficeController
  * @class OfficeController
  */
-class OfficesController {
```
Your controller should be in singular, not plural; *OfficeController*

amaechi-chuks replied:
Acknowledged with thanks.
I have implemented the changes based on the project naming convention.

yomigeek reviewed (changes requested):
Good job. However, implement requested changes.

yomigeek commented on server/app.js:
```diff
@@ -16,9 +16,15 @@ app.use(bodyParser.urlencoded({ extended: true }));
 
 app.all('*', (req, res) => res.status(404).json({
+  status: 404,
   message: 'Wrong endpoint. Such endpoint does not exist',
```
Please check the project document for the correct response message format for errors.

yomigeek replied:
Please check the project document for the correct response message format for errors.

yomigeek commented on server/controller/officeController.js:
```diff
@@ -1,10 +1,10 @@
  * @class OfficeController
  */
-class OfficesController {
+export default class OfficesController {
```
Please your class name should be singular; OfficeController. Also your controller file should be in Pascal casing. N:B the controller filename and the class name should be same based on the AirBnB guideline.

amaechi-chuks replied:
Acknowledged.
I have implemented the feedback.

yomigeek commented on server/app.js:
```diff
@@ -2,7 +2,7 @@ import express from 'express';
 import bodyParser from 'body-parser';
 import winston from 'winston';
-import routes from './routes/routes';
+import routes from './route/Route';
```
Your file name is in Pascal casing instead of camel casing. Only controller files use Pascal casing.

amaechi-chuks committed:
[chore] Implement LFA feedback on naming convention

yomigeek commented on server/middleware/validateParty.js:
```diff
@@ -1,31 +1,31 @@
+  static findPartyById(req, res, next) {
     const { id } = req.params;
     if (!Number(id)) {
       return res.status(400).json({
```
Status code should be 404 for non-existing values.

amaechi-chuks replied:
Acknowledged and resolved

yomigeek commented on server/middleware/validateParty.js:
```diff
@@ -101,12 +101,10 @@ class ValidateParties {
-    }
-    if (!logoUrl || logoUrl === undefined) {
+    } else if (!logoUrl || logoUrl === undefined) {
       error = 'Party  must be specified';
```
I think this should be Party logo must be specified instead.

amaechi-chuks replied:
Acknowledged and resolved

yomigeek commented:
change Route.js to route.js (File name casing issues)

amaechi-chuks committed:
fix(UI template): Implement LFA feedback on UI template design (#33)

Add apply page for a user to participate as a candidate
Add a office record page for an admin to view
Add awesome font icons
[Finishes | Delivers #163375971]

amaechi-chuks committed:
fix(UI template): Implement LFA feedback on UI template design (#33)

Add apply page for a user to participate as a candidate
Add a office record page for an admin to view
Add awesome font icons
[Finishes | Delivers #163375971]
